sshd-core/src/test/java/org/apache/sshd/server/auth/AsyncAuthTestBase.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sshd.server.auth;
+
+import java.io.File;
+
+import com.jcraft.jsch.JSchException;
+
+import org.apache.sshd.common.FactoryManager;
+import org.apache.sshd.server.SshServer;
+import org.apache.sshd.server.auth.password.PasswordAuthenticator;
+import org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider;
+import org.apache.sshd.util.test.BaseTestSupport;
+import org.apache.sshd.util.test.EchoShellFactory;
+import org.junit.After;
+import org.junit.Test;
+
+/**
+ * @author <a href="mailto:dev@mina.apache.org">Apache MINA SSHD Project</a>
+ */
+public abstract class AsyncAuthTestBase extends BaseTestSupport {
+
+    SshServer server;
+
+    private PasswordAuthenticator authenticator;
+
+    public AsyncAuthTestBase() {
+    }
+
+    public void startServer() throws Exception {
+        startServer(null);
+    }
+
+    public void startServer(Integer timeout) throws Exception {
+        if (server != null) {
+            fail("Server already started");
+        }
+        server = SshServer.setUpDefaultServer();
+        if (timeout != null) {
+            server.getProperties().put(FactoryManager.AUTH_TIMEOUT, timeout.toString());
+        }
+        server.setPort(5000);
```

Comment:
Why do we have to use an explicit port and not let the server the 1st available one ? What if 5000 is used by the user's machine for something else ?